exercises/practice/bob/.approaches/when-expression/content.md

```diff
@@ -0,0 +1,60 @@
+# `when` expressions
+
+```kotlin
+object Bob {
+    fun hey(statement: String): String =
+        when {
+            statement.isQuestion() && statement.isYelling() -> "Calm down, I know what I'm doing!"
+            statement.isQuestion() -> "Sure."
+            statement.isYelling() -> "Whoa, chill out!"
+            statement.isSilence() -> "Fine. Be that way!"
+            else -> "Whatever."
+        }
+
+    private fun String.isSilence(): Boolean = this.isBlank()
+    private fun String.isQuestion(): Boolean = this.trim().endsWith('?')
+    private fun String.isYelling(): Boolean {
+        val letters = this.filter { it.isLetter() }
+        return if (letters.isEmpty())
+            false
+        else
+            letters.all { it.isUpperCase() }
+    }
+}
```

Comment:
What about?

```suggestion
    private fun String.isYelling() = any(Char::isLetter) && toUpperCase() == this
```

Reply:
Hey, 
looks good. Tried a solution using higher order functions but did not find an approach I was happy with.

I will update the solution using this one-liner, thanks.